src/server.js

```diff
@@ -378,23 +377,56 @@ function initializeClient(socket, client, token, lastMessage) {
 		client.unregisterPushSubscription(token);
 	});
 
-	socket.on("sign-out", () => {
-		delete client.config.sessions[token];
+	const sendSessionList = () => {
+		const sessions = _.map(client.config.sessions, (session, sessionToken) => ({
+			current: sessionToken === token,
+			active: _.find(client.attachedClients, (u) => u.token === sessionToken) !== undefined,
+			lastUse: session.lastUse,
+			ip: session.ip,
+			agent: session.agent,
+			token: sessionToken, // TODO: Ideally don't expose actual tokens to the client
+		}));
+
+		socket.emit("sessions:list", sessions);
+	};
+
+	socket.on("sessions:get", sendSessionList);
+
+	socket.on("sign-out", (tokenToSignOut) => {
+		// If no token provided, sign same client out
+		if (!tokenToSignOut) {
+			tokenToSignOut = token;
+		}
+
+		if (!(tokenToSignOut in client.config.sessions)) {
+			return;
+		}
+
+		delete client.config.sessions[tokenToSignOut];
 
 		client.manager.updateUser(client.name, {
 			sessions: client.config.sessions
-		}, (err) => {
-			if (err) {
-				log.error("Failed to update sessions for", client.name, err);
+		});
+
+		_.map(client.attachedClients, (attachedClient, socketId) => {
+			if (attachedClient.token !== tokenToSignOut) {
+				return;
 			}
+
+			const socketToRemove = manager.sockets.of("/").connected[socketId];
+
+			socketToRemove.emit("sign-out");
+			socketToRemove.disconnect();
 		});
 
-		socket.emit("sign-out");
+		sendSessionList();
```

Comment:
We'll want to wrap this in a check for logging out the current session, otherwise we'll try to send the session list to the session we just disconnected.

Reply:
Fixed.